src/pages/work/[slug].tsx

```diff
@@ -32,68 +45,111 @@ export async function getStaticProps(context: { params: { slug: string } }) {
     return match ? parseInt(match[0], 10) : 0;
   }
 
-  const number = getNumberFromString(slug);
+  const id = getNumberFromString(slug);
+  const nextId = (id + 1) % images.length;
   return {
     props: {
-      data: { slug, image: images[number], number },
+      data: {
+        id,
+        slug,
+        image: images[id],
+        nextProject: {
+          id: nextId,
+          image: images[nextId],
+        },
+      },
     },
   };
 }
 
-export default function Work({
-  data,
-}: InferGetStaticPropsType<typeof getStaticProps>) {
+interface Props extends InferGetStaticPropsType<typeof getStaticProps> {}
+
+export default function Work({ data }: Props) {
   const { state } = useGlobalState();
+  const nextProjectHref = `/work/item-${data.nextProject.id}`;
+  const [imgSrc, setImgSrc] = useState(data.image);
+  const [title, setTitle] = useState(`/work/${data.slug}`);
+  const [nextRoute, setNextRoute] = useState<string | null>(null);
+  const [hideContent, setHideContent] = useState(false);
+  const [hideFooter, setHideFooter] = useState(false);
 
   async function handleClick(event: MouseEvent<HTMLAnchorElement>) {
-    const href = event.currentTarget.getAttribute("href");
+    const { currentTarget } = event;
+    const href = currentTarget.getAttribute("href");
+
+    setNextRoute(href);
 
     event.preventDefault();
 
-    await animate(document.documentElement.scrollTop, 0, {
+    const rect = event.currentTarget.getBoundingClientRect();
```

Comment:
Getting the position of the hero's offset from the top of the window could be a way around the next js page scroll jump?

Reply:
So, do you mean offset the y position of the incoming hero, to match the outgoing footer?

Reply:
Could be one approach!